chatserver/src/main/java/pl/sages/javadevpro/Server.java

```diff
@@ -39,4 +47,47 @@ public void closeServerSocket() {
         }
     }
 
+    public void commandsHandle(String command, ChatUser user) {
+        String[] commandParts = command.split(" ");
+        switch (commandParts[0]) {
+            case "/join": { handleJoinCommand(commandParts[1], user); break; }
+            case "/exit": { handleExitCommand(user); break; }
+            default:
+                System.out.println("Command not supported");
+        }
+    }
+
+    public void handleJoinCommand(String chatName, ChatUser user) {
+        user.getRoom().removeChatUser(user);
+        ChatRoom chatRoom = rooms.stream()
+            .filter(r -> r.getChatRoomName().equals(chatName))
+            .findFirst().orElseGet(() -> {
+                ChatRoom room = new ChatRoom(chatName);
+                rooms.add(room);
+                return room;
+            });
+        chatRoom.addChatUser(user);
+        user.setRoom(chatRoom);
+    }
+
+    public void handleExitCommand(ChatUser user) {
+        String chatName = user.getRoom().getChatRoomName();
+        final String mainChatName = "#general";
+
+        if(chatName.equals(mainChatName))
+            return;
+
+        user.getRoom().removeChatUser(user);
+
+        ChatRoom mainChatRoom = rooms.stream()
+                .filter(r -> r.getChatRoomName().equals(mainChatName))
+                .findFirst().orElseGet(() -> {
```

Comment:
General chat room is created each time at server start. I think we don't need to check again if it's present?

Reply:
suggestion accepted